File: TradeHarborApi/Models/Post/TradePost.cs
using TradeHarborApi.Models.PostFeatures;

namespace TradeHarborApi.Models.Post
{
    /// <summary>
    /// Represents a trade post entity.
    /// </summary>
    public class TradePost
    {
        /// <summary>
        /// Gets or sets the user ID associated with the trade post.
        /// </summary>
        public string UserId { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the unique identifier of the trade post.
        /// </summary>
        public string TradeId { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the ticker symbol associated with the trade post.
        /// </summary>
        public string Ticker { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the position related to the trade post.
        /// </summary>
        public string Position { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the option related to the trade post.
        /// </summary>
        public string Option { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the strike price associated with the trade post.
        /// </summary>
        public string Strikeprice { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets any additional comments or notes related to the trade post.
        /// </summary>
        public string Comment { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the timestamp when the trade post was created.
        /// </summary>
        public DateTime Timestamp { get; set; }

        /// <summary>
        /// Gets or sets the first name of the user associated with the trade post.
        /// </summary>
        public string FirstName { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the last name of the user associated with the trade post.
        /// </summary>
        public string LastName { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the username of the user associated with the trade post.
        /// </summary>
        public string Username { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the URL of the user's profile picture.
        /// </summary>
        public string ProfilePictureUrl { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the number of votes received on the trade post.
        /// </summary>
        public int Votes { get; set; }

        /// <summary>
        /// Gets or sets the user's reaction to the trade post.
        /// </summary>
        public string UserReaction { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the collection of comments associated with the trade post.
        /// </summary>
        public IEnumerable<PostComment> Comments { get; set; } = [];
    }
}

File: TradeHarborClient/src/app/models/api-responses/tradePost.ts
import { PostComment } from "./postComment";

/**
 * Represents a trade post with user and comment information.
 */
export class TradePost {
  /**
   * The unique identifier for the user who created the trade post.
   */
  userId: string = '';

  /**
   * The unique identifier for the trade post.
   */
  tradeId: string = '';

  /**
   * The ticker symbol associated with the trade post.
   */
  ticker: string = '';

  /**
   * The position associated with the trade post.
   */
  position: string = '';

  /**
   * The type of option associated with the trade post.
   */
  option: string = '';

  /**
   * The strike price associated with the trade post.
   */
  strikeprice: string = '';

  /**
   * Comments about the trade post from poster.
   */
  comment: string = '';

  /**
   * The timestamp when the trade post was created.
   */
  timestamp: string = '';

  /**
   * The first name of the user who created the trade post.
   */
  firstName: string = '';

  /**
   * The last name of the user who created the trade post.
   */
  lastName: string = '';

  /**
   * The username of the user who created the trade post.
   */
  username: string = '';

  /**
   * The URL of the user's profile picture.
   */
  profilePictureUrl: string = '';

  /**
   * The number of votes received on the trade post.
   */
  votes: number = 0;

  /**
   * The user's reaction towards the trade post.
   */
  userReaction: string = '';

  /**
   * The comments associated with the trade post.
   */
  comments: PostComment[] = [];
}
